pyscriptjs/src/utils.ts

```diff
@@ -109,3 +111,35 @@ export function getAttribute(el: Element, attr: string): string | null {
     }
     return null;
 }
+
+export function calculatePaths(fetch_cfg: FetchConfig[]) {
+    const fetchPaths: string[] = [];
+    const paths: string[] = [];
+    fetch_cfg.forEach(function (each_fetch_cfg: FetchConfig) {
+        const from = each_fetch_cfg.from || "";
+        const to_folder = each_fetch_cfg.to_folder || ".";
+        const files = each_fetch_cfg.files;
+        if (files !== undefined)
+        {
+            for (const each_f of files)
+            {
+                const each_fetch_path = [from.endsWith('/') ? from.slice(0, -1) : from, each_f].filter(it => it!== "").join('/');
```

Comment:
I have no idea what this line does :). Sure, I could probably spend 5 minutes trying to understand it, OR you could write a nice function. If I understand correctly, there you want to do essentially the equivalent of `os.path.join`, don't you?
If so, I suggest to just write a small function `joinPaths` which does just this.
Bonus point: as soon as you have your own function, you can unit-test it very well, with all sort of corner cases.

Also: why do you have `filter(it => it !== "")`? Does it mean that if I have `files = ["a", "", "b"]`, the 2nd is silently ignored and it ends up fetching only 3 files?
If so, -1. It should be an error.

Reply:
done, added testing, the part about `filter(it => it !== "")` is needed since the behaviour of
`os.path.join('', '', '')` is simply `''` so we are replicating that in some sense.